include/spdk/nvme.h

```diff
@@ -1141,6 +1142,36 @@ int spdk_nvme_ns_cmd_reservation_report(struct spdk_nvme_ns *ns,
 					void *payload, uint32_t len,
 					spdk_nvme_cmd_cb cb_fn, void *cb_arg);
 
+/**
+ * \brief Sends an abort command to the Nvme device for the request indicated
+ *  by cid and sqid
+ *
+ * \param ctrlr controller to the Nvme device
+ * \param cid identification number of the command to be aborted.
+ * \param sqid submission queue number of the command to be aborted
+ * \param cb_fn a function pointer to the abort call back function
+ * \param cb_arg user argument to the abort command callback function
+ *
+ * \return 0 if successfully submitted, Less than 0 if it fails
+ *
+ * Note: The user must ensure to decrement the controller member
```

Comment:
Hi Daniel, there were two design alternatives if one chooses not to exceed the abort count limit:
1. the current implementation which places responsibility on the caller of 
    spdk_nvme_ctrlr_cmd_abort(..) to decrement  ctrlr->curr_abort_count 
    member in their abort callback function. 

2. The 2nd implementation would require the incrementation of  ctrlr->curr_abort_count to be
    localized as follows:
   a. acquire the lock before calling the abort callback function.
   b. If abort was performed in the abort callback function AND if the call to abort was successful,
       return a special code to indicate as such AND in order to increment the 
       ctrlr->curr_abort_count
       member. Otherwise, do not ctrlr->curr_abort_count member when abort callback function
       return.
   c. release the lock

   The 2nd implementation places no responsibility on the caller of spdk_nvme_ctrlr_cmd_abort(..)
   to decrement ctrlr->curr_abort_count member. However, there are two disadvantages, 1) the 
   callback function must return three different return types indicate that abort was called
   and the call succeeded, failed, or a controller reset was performed instead. 2) the lock will be
   acquired again in the abort function.

The 2nd choice makes the code reading difficult to understand so, we chose the current implementation.